pallets/qa-tools/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+// private-net to make circular dependencies work
+#[cfg(all(test, feature = "private-net", feature = "ready-to-test"))] // order-book
+mod tests;
+
+pub mod weights;
+use order_book::OrderBookId;
+pub use weights::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use common::prelude::FixedWrapper;
+    use common::{
+        balance, AssetInfoProvider, AssetName, AssetSymbol, Balance, BalancePrecision,
+        ContentSource, Description, PriceVariant,
+    };
+    use frame_support::dispatch::DispatchErrorWithPostInfo;
+    use frame_support::sp_runtime::traits::Zero;
+    use frame_support::traits::{Get, Time};
+    use frame_support::{dispatch::PostDispatchInfo, pallet_prelude::*};
+    use frame_system::pallet_prelude::*;
+    use order_book::DataLayer;
+    use order_book::{MomentOf, OrderBook};
+    use sp_std::prelude::*;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + order_book::Config + trading_pair::Config {
+        type WeightInfo: WeightInfo;
+        type OrderBookOrderLifespan: Get<MomentOf<Self>>;
+        type AssetInfoProvider: AssetInfoProvider<
+            Self::AssetId,
+            Self::AccountId,
+            AssetSymbol,
+            AssetName,
+            BalancePrecision,
+            ContentSource,
+            Description,
+        >;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Order book related errors
+        OrderBook(OrderBookError),
+    }
+
+    #[derive(Encode, Decode, scale_info::TypeInfo, frame_support::PalletError)]
+    pub enum OrderBookError {
+        /// Order book does not exist for this trading pair
+        UnknownOrderBook,
+    }
+
+    #[derive(
+        Encode,
+        Decode,
+        Eq,
+        PartialEq,
+        Copy,
+        Clone,
+        PartialOrd,
+        Ord,
+        RuntimeDebug,
+        Hash,
+        scale_info::TypeInfo,
+        MaxEncodedLen,
+    )]
+    pub struct OrderBookFillSettings {
+        /// Best (highest) price for placed buy orders
+        pub best_bid_price: order_book::types::OrderPrice,
+        /// Best (lowest) price for placed sell orders
+        pub best_ask_price: order_book::types::OrderPrice,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Create multiple order books with default parameters (if do not exist yet).
+        ///
+        /// Parameters:
+        /// - `origin`: caller, should be account because error messages for unsigned txs are unclear,
+        /// - `order_book_ids`: ids of the created order books; trading pairs are created
+        /// if necessary,
+        #[pallet::call_index(0)]
+        #[pallet::weight(<T as Config>::WeightInfo::order_book_create_empty_many())]
+        pub fn order_book_create_empty_many(
```

Comment:
```suggestion
        pub fn create_empty_order_books_batch(
```

Reply:
The idea is to separate all extrinsics for order books to distinguish them with addition of other callables into this pallet